File: Checkout/Java/get_order.java
import java.net.HttpURLConnection;
import java.net.URL;
import java.net.URI;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.text.SimpleDateFormat;
import java.util.Base64;
import java.util.Date;
import java.util.HashMap;
import java.util.Map;
import java.nio.file.Files;
import java.nio.file.Paths;

public class get_order {

    private static String order_id = "";
    private static final String MERCHANT_ID = "CHECKOUT_MERCHANT_ID";
    private static final String SECRET_WORD = "CHECKOUT_SECRET_KEY";
    //private static final String BASE_URL = "https://paymentadminapistage.svea.com/api/v1/orders/";
    private static final String BASE_URL = "https://checkoutapistage.svea.com/api/orders/";
    private static final String CONTENT_TYPE = "application/json";

    public static void main(String[] args) {
        System.out.println("Running GET request for Checkout (Java)");
        try {
            try {
                order_id = new String(Files.readAllBytes(Paths.get("./created_order_id.txt"))).trim();
                //System.out.println("Using OrderId: " + order_id);
            } catch (IOException e) {
                System.err.println("Error reading OrderId from file: " + e.getMessage());
                return;
            }

            String url = BASE_URL + order_id;
            Map<String, String> myHeaders = getRequestHeaders("", null);

            //URL obj = new URL(url);
            URL obj = URI.create(url).toURL();
            HttpURLConnection con = (HttpURLConnection) obj.openConnection();
            con.setRequestMethod("GET");

            for (Map.Entry<String, String> entry : myHeaders.entrySet()) {
                con.setRequestProperty(entry.getKey(), entry.getValue());
            }

            int responseCode = con.getResponseCode();
            //System.out.println("Response Code : " + responseCode);
            if (responseCode == 200)
                System.out.println("Success!");
            else
                System.out.println("Failed...");

            if (responseCode == HttpURLConnection.HTTP_OK) {
                BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
                String inputLine;
                StringBuffer response = new StringBuffer();

                while ((inputLine = in.readLine()) != null) {
                    response.append(inputLine);
                }
                in.close();

                //System.out.println(response.toString());
            } else {
                System.out.println("Error: Request returned non-OK status code");
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static Map<String, String> getRequestHeaders(String requestBody, Map<String, String> extraHeaders) {
        String timestamp = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(new Date());
        String token = getAuthToken(timestamp, requestBody);
        Map<String, String> headers = new HashMap<>();
        headers.put("Authorization", "Svea " + token);
        headers.put("Timestamp", timestamp);

        if (extraHeaders != null) {
            headers.putAll(extraHeaders);
        }

        return headers;
    }

    public static String getAuthToken(String timestamp, String requestBody) {
        String digest = requestBody + SECRET_WORD + timestamp;
        MessageDigest md;
        try {
            md = MessageDigest.getInstance("SHA-512");
            byte[] hashed = md.digest(digest.getBytes());
            String auth = MERCHANT_ID + ":" + bytesToHex(hashed);
            return Base64.getEncoder().encodeToString(auth.getBytes());
        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
            return null;
        }
    }

    private static String bytesToHex(byte[] bytes) {
        StringBuilder sb = new StringBuilder();
        for (byte b : bytes) {
            sb.append(String.format("%02x", b));
        }
        return sb.toString();
    }
}

File: Checkout/Go/get_order.go
package main

import (
	"crypto/sha512"
	"encoding/base64"
	"fmt"
	"net/http"
	"time"
    "io"
    "strings"
)

const (
	orderID     = "CHECKOUT_ORDER_TO_FETCH"
	merchantID  = "CHECKOUT_MERCHANT_ID"
	secretWord  = "CHECKOUT_SECRET_KEY"
	//baseURL     = "https://paymentadminapistage.svea.com/api/v1/orders/"
    baseURL     = "https://checkoutapistage.svea.com/api/orders/"
	contentType = "application/json"
)

type TestClass struct{}

func (tc *TestClass) getRequestHeaders(requestBody string, extraHeaders map[string]string) map[string]string {
	timestamp := time.Now().UTC().Format("2006-01-02 15:04:05")
	token := tc.getAuthToken(timestamp, requestBody)
	headers := map[string]string{
		"Authorization": fmt.Sprintf("Svea %s", token),
		"Timestamp":     timestamp,
	}

	if extraHeaders != nil {
		for key, value := range extraHeaders {
			headers[key] = value
		}
	}

	return headers
}

func (tc *TestClass) getAuthToken(timestamp, requestBody string) string {
	digest := requestBody + secretWord + timestamp
	h := sha512.New()
	h.Write([]byte(digest))
	hashed := h.Sum(nil)
	auth := fmt.Sprintf("%s:%x", merchantID, hashed)
	token := base64.StdEncoding.EncodeToString([]byte(auth))
	return token
}

func main() {
    fmt.Println("Running GET request for Checkout (Go)")
	testInstance := &TestClass{}
	myHeaders := testInstance.getRequestHeaders("", nil)
	//fmt.Println(myHeaders)

	url := baseURL + orderID
	// payload := map[string]interface{}{}

	//fmt.Println("Fetching order:", orderID)
	client := &http.Client{}
	req, err := http.NewRequest("GET", url, nil)
	if err != nil {
		fmt.Println("Error creating request:", err)
		return
	}

	for key, value := range myHeaders {
		req.Header.Set(key, value)
	}

	resp, err := client.Do(req)
	if err != nil {
		fmt.Println("Error making request:", err)
		return
	}
	defer resp.Body.Close()

	//fmt.Println("Response Status:", resp.Status)

    if resp.StatusCode == http.StatusOK {
        bodyBytes, err := io.ReadAll(resp.Body)
        if err != nil {
            fmt.Println("Error")
        }
        bodyString := string(bodyBytes)
        //fmt.Println(bodyString)
        if resp.StatusCode == 200 && strings.Contains(strings.ToLower(bodyString), "iframe") {
            fmt.Println("Success!")
        } else {
            fmt.Println("Failed...")
        }
    }

	if resp.StatusCode != http.StatusOK {
		fmt.Println("Error: Request returned non-OK status code")
		return
	}
}
